src/stores/map.ts

```diff
@@ -117,6 +133,124 @@ export const useMapStore = defineStore("map", () => {
     }
   }
 
+  /**
+   * Applies the given offset to the input coordinates and returns the updated coordinates.
+   * @param coordinates - The input coordinates as an array of [x, y].
+   * @param offset - The offset to be applied as an array of [x, y].
+   * @returns The updated coordinates after applying the offset.
+   */
+  function applyOffsetToCoordinates(coordinates: [number, number], offset: [number, number]): [number, number] {
+    return [coordinates[0] + offset[0], coordinates[1] + offset[1]];
+  }
+
+  /**
+   * Updates the offset for the given buildingSlug with the specified new offset.
+   * @param buildingSlug - The building identifier.
+   * @param newOffset - The new offset to be applied as an array of [x, y].
+   */
+  function updateOffsetForBuildingSlug(buildingSlug: string, newOffset: [number, number]) {
+    if (clusterCoordinates.value && clusterCoordinates.value[buildingSlug]) {
+      clusterCoordinates.value[buildingSlug] = applyOffsetToCoordinates(clusterCoordinates.value[buildingSlug], newOffset);
+    }
+  }
+
+  /**
+   * A collection of initial cluster coordinates indexed by building slugs.
+   */
+  const initialClusterCoordinates: { [key: string]: [number, number] } = {
```

Comment:
Maybe give these a more descriptive name, as this now seem like hardcoded coordinates.